addons/pkg/util/cluster_variable_util.go

```diff
@@ -29,6 +30,41 @@ func ParseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName
 	return result.(string), err
 }
 
+func ParseClusterVariableInterface(cluster *clusterapiv1beta1.Cluster, variableName, keyName string) (string, error) {
+	var result interface{}
+
+	result, err := parseClusterVariable(cluster, variableName)
+	if err != nil || result == nil {
+		return "", err
+	}
+	interfaceVars := result.(map[string]interface{})
+
+	if valueName, ok := interfaceVars[keyName]; ok {
+		return valueName.(string), err
+    }
+	return "", err
+}
+
+func ParseClusterVariableList(cluster *clusterapiv1beta1.Cluster, variableName, keyName, data string) (string, error) {
+	var result interface{}
+    var sb strings.Builder
+
+	result, err := parseClusterVariable(cluster, variableName)
+	if err != nil || result == nil {
+		return "", err
+	}
+	interfaceVars := result.(map[string]interface{})
+	if valueList, ok := interfaceVars[keyName]; ok {
+		rec := valueList.([]interface{})
```

Comment:
Same here, we need to check if `valueList.([]interface{})` conversion is valid.

Reply:
added type check before return